app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt

```diff
@@ -178,4 +215,54 @@ fun searchScreen(interact: SearchRobot.() -> Unit): SearchRobot.Transition {
     return SearchRobot.Transition()
 }
 
+private fun assertKeyboardVisibility(isExpectedToBeVisible: Boolean) = {
+    mDevice.waitNotNull(
+        Until.findObject(
+            By.text("Search Engine")
+        ), waitingTime
+    )
+    assertEquals(
+        isExpectedToBeVisible,
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+            .executeShellCommand("dumpsys input_method | grep mInputShown")
+            .contains("mInputShown=true")
+    )
+}
+
+private fun assertSearchEngineList() {
+    scrollToElementByText("Google")
+    onView(withText("Google"))
+        .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+    scrollToElementByText("Amazon.com")
+    onView(withText("Amazon.com"))
+        .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+    scrollToElementByText("Bing")
+    onView(withText("Bing"))
+        .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+    scrollToElementByText("DuckDuckGo")
+    onView(withText("DuckDuckGo"))
+        .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+    scrollToElementByText("Twitter")
+    onView(withText("Twitter"))
+        .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+    scrollToElementByText("Wikipedia")
+    onView(withText("Wikipedia"))
+        .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+}
+
+private fun assertSearchEngineIconsList(expectedText: String) =
+    onView(allOf(withId(R.id.mozac_browser_awesomebar_icon), hasSibling(withText(expectedText))))
+        .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+
+private fun selectDefaultSearchEngine(searchEngine: String) {
+    onView(withText(searchEngine))
```

Comment:
To be sure we can select each searchEngine in the list, let's dismiss the keyboard first by doing:
` onView(withId(R.id.mozac_browser_toolbar_edit_icon)).click()`
I have found many issues when running the test to be able to click on Twitter or the next search Engine in the list. With this line we are sure all the search engines in the list are visible

Reply:
(This would go in the previous line to this one, just below the method definition)